### PR TITLE
Feature/context and portals

### PR DESCRIPTION
**Pull request by franmolmedo**

### CODE REVIEW

File: franmolmedo/react-context-and-portals/readme.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+Para hacer las pruebas vamos a utilizar la última versión disponible de React, en este momento (16.03.alpha1). Para ello, partimos de [create-react-app](https://github.com/facebook/create-react-app), el starter amparado por Facebook para aplicaciones React que nos monta la estructura básica de la aplicación. Nos vamos a nuestra consola y creamos la aplicación de ejemplo:
+
+```javascript
+npx create-react-app react-context-portals-example
+```
+
+![create-react-app](images/create-react-app.png)
+
+Nos situamos en la carpeta que nos ha creado y abrimos la aplicación en nuestro editor favorito. Create React App nos crea una aplicación React bastante estándar.
```
Comment (danimart1991) [inline franmolmedo/react-context-and-portals/readme.md]: Create React App -> \`Create React App\`

File: franmolmedo/react-context-and-portals/readme.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
+![Header](images/react.png)
```
Comment (danimart1991) [inline franmolmedo/react-context-and-portals/readme.md:1]: Add **Bold** and *Italic* along the post.

File: franmolmedo/react-context-and-portals/readme.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    "scripts": {
+        "start": "react-scripts start",
+        "build": "react-scripts build",
+        "test": "react-scripts test --env=jsdom",
+        "eject": "react-scripts eject"
+    }
+}
+```
+
+Una vez hecho esto, nos vamos a la consola y ejecutamos **_yarn start_** para que se nos lance la aplicación en nuestro navegador:
```
Comment (danimart1991) [inline franmolmedo/react-context-and-portals/readme.md]: Change \**\__yarn start__** for \`yarn start\`

File: franmolmedo/react-context-and-portals/readme.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+                <button onClick={() => this.decrement()}>DECREMENT</button>
+            </Fragment>
+        );
+    }
+}
+
+export default App;
+```
+
+Tenemos el estado que definimos con el valor inicial de 0 dentro de la propiedad current y, posteriormente, declaramos dos funciones, una para incrementar este valor y otra para decrementarlo, seteando el nuevo valor en el state. Posteriormente, tenemos un par de botones que invocan a estas funciones en el evento click y el componente hijo **Counter** que se encarga de mostrar por pantalla el valor actual del contador. Este componente consistiría realmente en un componente funcional muy sencillo que recibe como prop **_currentValue_** y la muestra por pantalla.
```
Comment (danimart1991) [inline franmolmedo/react-context-and-portals/readme.md]: ¿seteando?

**Review by danimart1991 — APPROVED** [review]
Good post. Request some minor style changes.

File: franmolmedo/react-context-and-portals/readme.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    };
+
+    increment = () => this.setState({ current: this.state.current + 1 });
+    decrement = () => this.setState({ current: this.state.current - 1 });
+
+    render() {
+        const value = {
+            state: this.state,
+            actions: {
+                increment: this.increment.bind(this),
```
Comment (IvanRodriCalleja) [inline franmolmedo/react-context-and-portals/readme.md:178]: why not use here
actions: {
    increment: () => this.increment(),
    decrement: () => this.decrement()
}

like previous steps?

**Review by IvanRodriCalleja — APPROVED** [review]
Interesting post with the new react features.

For next post i would like a post about testing in react and good practise for testing

File: franmolmedo/react-context-and-portals/readme.md (hunk 1 above)
Comment (franmolmedo) [reply franmolmedo/react-context-and-portals/readme.md]: Done

File: franmolmedo/react-context-and-portals/readme.md (hunk 2 above)
Comment (franmolmedo) [reply franmolmedo/react-context-and-portals/readme.md:1]: It is already done

File: franmolmedo/react-context-and-portals/readme.md (hunk 3 above)
Comment (franmolmedo) [reply franmolmedo/react-context-and-portals/readme.md]: It is in italic

File: franmolmedo/react-context-and-portals/readme.md (hunk 4 above)
Comment (franmolmedo) [reply franmolmedo/react-context-and-portals/readme.md]: Changed

File: franmolmedo/react-context-and-portals/readme.md (hunk 5 above)
Comment (franmolmedo) [reply franmolmedo/react-context-and-portals/readme.md:178]: Showing the two ways to do the same.

**Pull request merged**